% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
  \usepackage{amssymb}
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={cILR: Taxonomic Enrichment Analysis with competitive Isometric Log Ratios},
  pdfauthor={Quang P. Nguyen, Dartmouth College; Anne G. Hoen, Dartmouth College; H. Robert Frost, Dartmouth College},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifluatex
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1 \everypar{\setlength{\hangindent}{\cslhangindent}}\ignorespaces\fi
  % set entry spacing
  \ifnum #2 > 0
  \setlength{\parskip}{#2\baselineskip}
  \fi
 }%
 {}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}

\title{cILR: Taxonomic Enrichment Analysis with competitive Isometric
Log Ratios}
\author{Quang P. Nguyen, Dartmouth College \and Anne G. Hoen, Dartmouth
College \and H. Robert Frost, Dartmouth College}
\date{}

\begin{document}
\maketitle

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

\hypertarget{materials-and-methods}{%
\section{Materials and Methods}\label{materials-and-methods}}

\hypertarget{methods}{%
\section{Methods}\label{methods}}

\hypertarget{competitive-isometric-log-ratio-cilr}{%
\subsection{Competitive Isometric Log-ratio
(cILR)}\label{competitive-isometric-log-ratio-cilr}}

The cILR method generates sample-specific enrichment scores for
microbial sets using the isometric log-ratio transformation {[}1{]}. The
cILR method takes two inputs:

\begin{itemize}
\tightlist
\item
  \(\mathbf{X}\): \(n\) by \(p\) matrix of positive counts for \(p\)
  taxa and \(n\) samples measured through either targeted sequencing
  \(p\) matrix of positive counts for \(p\) taxa and \(n\) samples
  measured through either targeted sequencing (such as 16S rRNA) or
  whole genome shotgun sequencing. Usually \(\mathbf{X}\) is generated
  from standard sequence processing pipelines such as DADA2 {[}2{]} and
  MetaPhlAn2 {[}3{]}.
\item
  \(\mathbf{A}\): \(p\) by \(m\) indicator matrix annotation the
  membership of each taxa \(p\) to \(m\) sets of interest. These sets
  can be Linnean taxonomic classifications annotated using databases
  such as SILVA {[}4{]} or those based on more functionally driven
  categories such as the functional tropism of microbes (\(a_{i,j} = 1\)
  indicates that microbe \(i\) belongs to set \(j\).
\end{itemize}

The cILR method generates one output:

\begin{itemize}
\tightlist
\item
  \(\mathbf{E}\): \(n\) by \(m\) matrix indicating the enrichment score
  of \(m\) pre-defined sets as specified in \(\mathbf{A}\) across \(n\)
  samples.
\end{itemize}

The procedure is as follows:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  \textbf{Compute the cILR statistic}: Let \(\mathbf{M}\) be a \(n\) by
  \(m\) matrix of cILR scores. Let \(\mathbf{M}_{i,k}\) be the be cILR
  scores for set \(k\) of sample \(i\):\\
  \[\mathbf{M}_{i,k} = \sqrt{\frac{\sum_k A_{ik}(p - \sum_k A_{ik})}{p}} \ln \left( \frac{g(\mathbf{X}_{i,j}|\mathbf{A}_{j,k} = 1)}{g(\mathbf{X}_{i,j}|\mathbf{A}_{j,k} \neq 1))} \right)\]
  where \(g()\) is the geometric mean. This represents the ratio of the
  geometric mean of the relative abundance of taxa assigned to set \(k\)
  and remainder taxa.
\item
  \textbf{Compute the cILR statistic on permuted \(\mathbf{X}\)}: We
  seek to evaluate the empirical null distribution of the cILR statistic
  under \(H_o\) that relative abundances in \(\mathbf{X}\) of members of
  set \(k\) are not enriched compared to those not in set \(k\). Let
  \(\mathbf{X}_p\) be the column permuted relative abundance matrix, and
  \(\mathbf{M}_p\) be the corresponding cILR scores generated from
  \(\mathbf{X}_p\).\\
\item
  \textbf{Fit Gaussian mixture distribution for each column of
  \(\mathbf{M}_p\)}
\item
  \textbf{Calculate finalized cILR scores as CDF values of the fitted
  mixture distribution}
\end{enumerate}

\hypertarget{results}{%
\section{Results}\label{results}}

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

\hypertarget{conclusion}{%
\section*{Conclusion}\label{conclusion}}
\addcontentsline{toc}{section}{Conclusion}

\hypertarget{refs}{}
\begin{CSLReferences}{0}{0}
\leavevmode\hypertarget{ref-egozcue2003}{}%
1. Egozcue JJ, Pawlowsky-Glahn V, Mateu-Figueras G, Barcelo-Vidal C.
Isometric {Logratio Transformations} for {Compositional Data Analysis}.
Mathematical Geology. 2003;22.

\leavevmode\hypertarget{ref-callahan2016}{}%
2. Callahan BJ, McMurdie PJ, Rosen MJ, Han AW, Johnson AJA, Holmes SP.
{DADA2}: {High}-resolution sample inference from {Illumina} amplicon
data. Nature Methods. 2016;13:581--3.

\leavevmode\hypertarget{ref-truong2015}{}%
3. Truong DT, Franzosa EA, Tickle TL, Scholz M, Weingart G, Pasolli E,
et al. {MetaPhlAn2} for enhanced metagenomic taxonomic profiling. Nature
Methods. 2015;12:902--3.

\leavevmode\hypertarget{ref-quast2013}{}%
4. Quast C, Pruesse E, Yilmaz P, Gerken J, Schweer T, Yarza P, et al.
The {SILVA} ribosomal {RNA} gene database project: Improved data
processing and web-based tools. Nucleic Acids Research. 2013;41:D590--6.

\end{CSLReferences}

\end{document}
