<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Taxonomic enrichment analysis using isometric log-ratios</title>
    <meta charset="utf-8" />
    <meta name="author" content="Quang Nguyen" />
    <meta name="date" content="2021-02-22" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/metropolis.css" rel="stylesheet" />
    <link href="libs/remark-css/metropolis-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="presentation.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">




class: title-slide, center, middle

# Taxonomic Enrichment Analysis with Isometric Log Ratios  

## Quang P. Nguyen 

### Advisors: Anne G. Hoen &amp; H. Robert Frost

### Department of Epidemiology &amp; Biomedical Data Science
### Dartmouth College

---

# Microbiome plays an important role in development 
.pull-left[.center-left[  
- The microbiome defined as the collective genomes of the microbes (composed of bacteria, bacteriophage, fungi, protozoa and viruses) that live inside and on the human body. [1]    
- Studies have shown associations between changes in the composition of the microbiome and exposures/health outcomes in early life [2]  
]]
.pull-right[
&lt;div class="figure"&gt;
&lt;img src="figures/infant_microbiota.jpg" alt="Backhed et al. 2015" width="95%" /&gt;
&lt;p class="caption"&gt;Backhed et al. 2015&lt;/p&gt;
&lt;/div&gt;
]


.footnote[
  [1] genome.gov   
  [2] Arrieta et al. 2014      
]
---
# Multi-omics approaches to study the microbiome  
.center[.middle[
&lt;img src="figures/meta_omics.png" width="100%" /&gt;
]]

---
# Microbial compositions are highly dynamic  

.center[
&lt;div class="figure"&gt;
&lt;img src="figures/stewart.png" alt="Stewart et al. 2018" width="50%" /&gt;
&lt;p class="caption"&gt;Stewart et al. 2018&lt;/p&gt;
&lt;/div&gt;

]

---
# Taxa-function relationships 
.pull-left[.center-left[
- Taxa-function relationships describe the distribution of functional roles across different taxonomic components [1]  

- Studies found complex ecological factors (such as nice partitioning and functional redundancy), affect the overall role of each taxa in shaping the host-microbe interaction [2]
]]
.pull-right[.center-right[
&lt;div class="figure"&gt;
&lt;img src="figures/borenstein_manor_2018.png" alt="Eng &amp;amp; Borenstein. 2018" width="100%" /&gt;
&lt;p class="caption"&gt;Eng &amp; Borenstein. 2018&lt;/p&gt;
&lt;/div&gt;
]]
.footnote[
[1] Eng &amp; Borenstein 2018  
[2] Vieira-Silva et al. 2016
]

---
# Towards functionally aware taxonomic analysis   
.center[
&lt;img src="figures/function_analysis_paradigm.png" width="60%" /&gt;
]

---
# Variable aggregation   
.pull-left[.center-left[
- Variable aggregation provide a way to group taxa by their common function  
- Aggregating variables can address issues of compositionality, high dimensionality, interpretability, and sparsity.  
- Traditional categories of aggregation in microbiome analyses are to higher taxonomic levels.  
]]

.pull-right[
&lt;img src="figures/hierarchy.jpg" width="80%" /&gt;

]

---
# Isometric log-ratio transformation  

.pull-left[
- All sequencing data (and especially microbiome) are **compositional**!  
  - Analyses should only be performed using **ratios between variables**    
  - Common transformations include the centered log ratio or the additive log ratio  

- The isometric log ratio transformation is a natural way to aggregate compositional variables  
`$$X_{R/S} = \sqrt{\frac{rs}{r+s}} \log \frac{g(R_i)}{g(S_i)}$$`  
  - Intuitively, if members of set `\(R\)` increases geometrically by a factor `\(\alpha\)` while member of set `\(S\)` is set to be constant, then the coordinate will change by addition of `\(\sqrt{\frac{rs}{r+s}} \log(\alpha)\)`    

]

.pull-right[.center-right[
&lt;div class="figure"&gt;
&lt;img src="figures/example_ilr.jpg" alt="Source: stackoverflow" width="80%" /&gt;
&lt;p class="caption"&gt;Source: stackoverflow&lt;/p&gt;
&lt;/div&gt;
]]

---
# cILR: Competitive isometric log-ratio transform
.center[
Competitive null hypothesis: the abundance of taxa within the set is not enriched compared to those outside of the set 

&lt;img src="figures/cilr_general.png" width="40%" /&gt;
]

---
# Workflow  
.center[
&lt;img src="figures/workflow.png" width="50%" /&gt;
]  




---
# Simulation Design  
.center2[
- We assumed that the distribution of microbiome counts follows a zero-inflated negative binomial distribution [1]  
`$$X_{ij} = \bigg\{\begin{array}{cc}0&amp;\text{with probability }p_{ij}\\NB(\mu_i,\gamma_i)&amp; \text{with probability } (1 - p_{ij}) \end{array}$$`
- To incorporate flexible correlation structures, we utilized the NorTA (Normal to Anything) approach [2]  
Given an `\(n\)` by `\(p\)` matrix of values `\(U\)` sampled from multivariate normal distribution with correlation matrix `\(\rho\)`, we can generate target microbiome count vector `\(X_i\)` for taxa `\(i\)` following distribution `\(P\)` characterized by the cumulative distribution `\(\mathbb{F}\)`:  
`$$X_i = \mathbb{F}^{-1}(\Phi_{U_i})$$`
where `\(\Phi_{U_i}\)` represent samples from the standard multivariate normal distribution.  ]

.footnote[
[1] Calgaro et al. 2020    
[2] Cairo et al. 1997  
]



---
# Null distribution for significant testing    

.pull-left[
&lt;div class="figure"&gt;
&lt;img src="../manuscript/figures/distr_density_plot_real.png" alt="Distribution fitting using null data from Human Microbiome Project" width="90%" /&gt;
&lt;p class="caption"&gt;Distribution fitting using null data from Human Microbiome Project&lt;/p&gt;
&lt;/div&gt;
]
.pull-right[
.center-right[
&lt;div class="figure"&gt;
&lt;img src="../manuscript/figures/kurtosis_skewness_sim.png" alt="Kurtosis and Skewness from simulated data" width="120%" /&gt;
&lt;p class="caption"&gt;Kurtosis and Skewness from simulated data&lt;/p&gt;
&lt;/div&gt;

]]

---
# Adjusting for inter-taxa correlation  
.pull-left[.center-left[
- However, permuting taxa disrupts the natural correlation structure, leading to underestimation of the variance of the permuted null.    
- Existing methods in the gene set testing literature accounts for this by estimating a variance inflation factor (VIF) [1]   
- Estimate this variance inflation by using the **estimated variance of the test statistic under unpermuted data**    
&lt;!--
&lt;img src="figures/three-steps.png" width="60%" /&gt;
--&gt;
]]

.pull-right[.center-right[
&lt;div class="figure"&gt;
&lt;img src="figures/p-value histogram.png" alt="Simulations under the global null" width="1092" /&gt;
&lt;p class="caption"&gt;Simulations under the global null&lt;/p&gt;
&lt;/div&gt;
]]

.footnote[
[1] Wu &amp; Smyth, 2012
]

---
# Goodness of fit  
.center[
&lt;img src="figures/gof_inter-taxa.png" width="45%" /&gt;
]

---
# Preliminary results  
.huge[
- We evaluated our method for three primary tasks:  
  - Single-sample significance testing  
  - Disease prediction  
  - Differential abundance analysis   
- Parametric simulations using the zero-inflated negative binomial distribution. We varied inter-taxa correlation (assumed to be exchangable), added sparsity, set size and effect size where appropriate  
- Real data evaluation  
]

---
# Single-sample significance testing   
.center[.middle[
&lt;img src="figures/prelim_simulations.png" width="1979" /&gt;
]]

---
# Single-sample significance testing  
.center[.middle[
&lt;img src="figures/data_single_sample.png" width="60%" /&gt;
]]
---
# Next steps  
.center2[.huge[
- Continue to refine our statistical model, including expanding to multiple-sample hypothesis testing, and variance inflation estimation procedure especially for mixture distributions  
- Evaluate model in all three outlined tasks across both parametric simulations and real data implementations ]]

---
# Acknowledgements  

.pull-left[
  **Thesis Advisors**  
  Anne G. Hoen, PhD    
  H. Robert Frost, PhD   
  **Thesis Committee**  
  Margaret R. Karagas, PhD  
  Brock C. Christensen, PhD  
  **Hoen Lab**  
  Weston Viles, PhD  
  Modupe O. Coker, PhD  
  Jie Zhou, PhD  
  Rebecca Lebeaux  
  Erika Dade  
  **Frost Lab**  
  Xingyu Zheng  
  Courtney Schiebout  
  **Research Funding Sources**  
  NLM R01LM012723 (AGH)  
  NLM K01LM012426 (HRF)  

]

.pull-right[
  **New Hampshire Birth Cohort Study**   
  Margaret K. Karagas, PhD (PI)  
  Juliette C. Madan, MD  
  All staff and participants  
  **UNC Metabolomics and Exposome Core**  
  Susan Sumner, PhD  
  Susan McRitchie, MS  
  Wilmal Pathmasiri, PhD  
  **Marine Biological Laboratory**  
  Hillary Morrison, PhD    
  **QBS Administration**  
  Diane Gilbert Diamond, ScD  
  Scott Gerber, PhD  
  Rosemary A. White  
  Shaniqua A. Jones  
  Kristine A. Griffin, PhD  
]




    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"ratio": "16:9",
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
