\documentclass{article}
\usepackage{amsfonts}
\usepackage{graphicx}
\usepackage[margin=1in]{geometry}
\usepackage{bm}
\usepackage{amsmath}
\usepackage{authblk}

\title{Taxonomic Enrichment Analysis with Isometric Log-Ratios}
\author[1,2]{Quang P. Nguyen}
\author[1,2]{Anne G. Hoen}
\author[1]{H. Robert Frost}
\affil[1]{Department of Biomedical Data Science, Geisel School of Medicine at Dartmouth College, Hanover, NH 03755, USA}
\affil[2]{Department of Epidemiology, Geisel School of Medicine at Dartmouth College, Hanover, NH 03755, USA}
\date{}                     %% if you don't need date to appear
\setcounter{Maxaffil}{0}
\renewcommand\Affilfont{\itshape\small}

\begin{document}
\maketitle

\begin{abstract}
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque arcu libero, suscipit sed enim nec, posuere eleifend nunc. Phasellus vitae augue orci. Sed vestibulum nisi id augue porta, a sagittis magna accumsan. Sed augue mi, venenatis sed fringilla nec, ultrices id ipsum. Suspendisse ac sapien eu mi laoreet fringilla. Nulla facilisi. Sed eget feugiat erat, et efficitur risus. Duis sit amet nulla at leo dignissim porta. Morbi nec ligula non sapien fringilla congue. Proin consequat volutpat nulla, eu convallis leo tempor in. Mauris elit sem, dignissim sit amet sapien sed, varius laoreet felis. Etiam elementum vulputate justo non malesuada. Suspendisse a libero id massa pellentesque convallis at in nibh. Ut nec consequat ante, vitae convallis dui. Ut eu pharetra nisi. 
\end{abstract}

\section*{Background}
\section*{Methods}
\section*{Simulation Analysis}
We chose to simulate data according to \cite{sohn2015}

\section*{Real data analysis} 
We benchmark type I error rate on 16S and WGS data from the Human Microbiome Project (HMP) obtained from the packages \emph{HMP16SData} (ver. 1.9.3) and \emph{curatedMetagenomicData} packages in R. For each data set, we filtered out samples with less than 1000 total number of reads, as well as taxa with a proportion of zeroes of 0.9 or more. We then randomly assigned each sample into one of two arbitrary groups.  
\newpage
\bibliography{tax_agg}{}
\bibliographystyle{plain}
\end{document}
\end{document}